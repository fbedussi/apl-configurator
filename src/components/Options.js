import React from 'react';
import {Link} from 'react-router-dom';


const Options = ({currentNode, options, setAnswer, baseUrl, history}) => {
    if (!options) {
        return null;
    }

    return <div className="question-options">
            {Object.keys(answers).map(key => {
                    return <button key={key} className="question-optionBtn ctaBtn" onClick={(e) => setAnswer(key)}>{answers[key]}</button>                    
                })
            }
    </div>
}

export default Options;