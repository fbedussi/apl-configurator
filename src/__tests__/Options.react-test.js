import React from 'react';
import Options from '../components/Options';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';   

const id = 0;
const options = ["no", "sÃ¬"];
const setAnswer = jest.fn();

test('Options component', () => {
  const component = renderer.create(
    <Options
        nodeId={id}
        options={options}
        setAnswer={setAnswer}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Options component - call handler when on option si clicked', () => {
  const component = shallow(
    <Options
        nodeId={id}
        options={options}
        setAnswer={setAnswer}
    />
  );

  component.find('#0_0').simulate('click');
  expect(setAnswer).toBeCalled();
});