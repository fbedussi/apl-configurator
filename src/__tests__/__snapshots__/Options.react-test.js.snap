// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options component 1`] = `
<div
  className="question-options"
>
  <button
    className="question-optionBtn ctaBtn"
    id="0_0"
    onClick={[Function]}
  >
    no
  </button>
  <button
    className="question-optionBtn ctaBtn"
    id="0_1"
    onClick={[Function]}
  >
    s√¨
  </button>
</div>
`;
